content/docs/hooks-faq.md

```diff
@@ -517,50 +823,50 @@ function Image(props) {
     return newObserver;
   }
 
-  // When you need it, call getObserver()
+  // عندما تحتاج إليه getObserver() استدعي
   // ...
 }
 ```
 
-This avoids creating an expensive object until it's truly needed for the first time. If you use Flow or TypeScript, you can also give `getObserver()` a non-nullable type for convenience.
+بذلك، تتجنب إنشاء كائنات مستهلكة للأداء حتى الحاجة الماسة إليها للمرة الأولى. إن كنت تستعمل Flow أو TypeScript، تستطيع أيضًا أن تعطي `getObserver()`‎ نوعًا غير معدوم (non-nullable type) للسهولة.
 
 
-### Are Hooks slow because of creating functions in render? {#are-hooks-slow-because-of-creating-functions-in-render}
+### هل تتسم الخطافات بالبطئ لإنشائها دوالًا في عملية التصيير؟{#are-hooks-slow-because-of-creating-functions-in-render}
 
-No. In modern browsers, the raw performance of closures compared to classes doesn't differ significantly except in extreme scenarios.
+لا. في المتصفحات الحديثة، لا يختلف الأداء الصافي (raw performance) للمغلفات (closures) بموازنته مع الأصناف اختلافًا كبيرًا باستثناء الحالات المبالغ بها.
 
-In addition, consider that the design of Hooks is more efficient in a couple ways:
+إضافةً لذلك، يعدُّ تصميم الخطافات أكثر فعالية من ناحيتين هما:
 
-* Hooks avoid a lot of the overhead that classes require, like the cost of creating class instances and binding event handlers in the constructor.
+* تخلصت الخطافات من الكثير من الأعباء التي تطلبها الأصناف مثل عبء طلب إنشاء نُسخٍ للصنف وربط معالجات حدث بالباني.
 
-* **Idiomatic code using Hooks doesn't need the deep component tree nesting** that is prevalent in codebases that use higher-order components, render props, and context. With smaller component trees, React has less work to do.
+* **لا تحتاج الشيفرة الاصطلاحية (Idiomatic code) التي تستعمل الخطافات إلى التشعب العميق لشجرة المكونات (deep component tree nesting)**  السائد في قواعد الشيفرة (codebases) التي تستعمل المكوانات ذات الترتيب الأعلى، وخاصييات التصيير، والسياق. مع شجرة مكونات صغيرة، يكون لدى React القليل من العمل لإنجازه.
 
-Traditionally, performance concerns around inline functions in React have been related to how passing new callbacks on each render breaks `shouldComponentUpdate` optimizations in child components. Hooks approach this problem from three sides.
+تقليديًّا، المخاوف التي تدور حول الدوال السطرية (inline functions) وتأثيرها على الأداء في React تتعلق بكيفية تمرير ردود النداء الجديدة في كل تصيير يفصل تحسينات `shouldComponentUpdate` في المكونات الأبناء. تتعامل الخطافات مع هذه المشكلة من ثلاث نواحٍ هي:
 
-* The [`useCallback`](/docs/hooks-reference.html#usecallback) Hook lets you keep the same callback reference between re-renders so that `shouldComponentUpdate` continues to work:
+* يمكِّنك الخطاف [`useCallback`](/docs/hooks-reference.html#usecallback) من الإبقاء على مرجع رد النداء نفسه بين عمليات إعادة التصيير، لذا يستمر `shouldComponentUpdate` بالعمل:
 
     ```js{2}
-    // Will not change unless `a` or `b` changes
+	// `b` أو `a` لن يتغير إلا إذا تغير
```

Comment:
This line is causing a formatting problem. Please [check it](https://deploy-preview-69--ar-reactjs.netlify.com/docs/hooks-faq.html#are-hooks-slow-because-of-creating-functions-in-render). We must find a solution for it

Reply:
it's looking good for me